src/TestProblem/Hydro/ClusterMerger/Flu_ResetByUser_ClusterMerger.cpp

```diff
@@ -0,0 +1,1132 @@
+   bool AdjustPosNow = false;
+   bool AdjustVelNow = false;
+
+// only adjust the BHs on the current maximum level
+   if ( CurrentMaxLv  &&  AdjustCount < int(TimeNew/AdjustPeriod) )
+   {
+      if ( AdjustBHPos == true )   AdjustPosNow = true;
+      if ( AdjustBHVel == true )   AdjustVelNow = true;
+      AdjustCount += 1;
+   } // if ( CurrentMaxLv  &&  AdjustCount < int(TimeNew/AdjustPeriod))
+   GetClusterCenter( lv, AdjustPosNow, AdjustVelNow, CM_BH_Pos, CM_ClusterCen, CM_BH_Vel );
+
+
+// (2) decide whether to merge BHs
+   const double AbsRelPos = DIST_3D_DBL( CM_BH_Pos[0], CM_BH_Pos[1] );
+   const double AbsRelVel = DIST_3D_DBL( CM_BH_Vel[0], CM_BH_Vel[1] );
+   const double soften    = amr->dh[MAX_LEVEL];
+   double escape_vel;
+   if ( AbsRelPos > soften )
+   {
+      escape_vel = sqrt( 2 * NEWTON_G * (CM_BH_Mass[0]+CM_BH_Mass[1]) / AbsRelPos );
+   }
+   else
+   {
+      escape_vel = sqrt( 2 * NEWTON_G * (CM_BH_Mass[0]+CM_BH_Mass[1]) / soften );
+   } // if ( AbsRelPos > soften ) ... else ...
+
+// merge the two BHs if they are located within R_acc, and the relative velocity is small enough
+   if ( Merger_Coll_NumBHs == 2 )
+   {
+      if ( AbsRelPos < R_acc  &&  AbsRelVel < 3*escape_vel )
+      {
+         Merger_Coll_NumBHs -= 1;
+         if ( CM_BH_Mass[0] >= CM_BH_Mass[1] )   merge_index = 1;   // record BH 1 merge BH 2 / BH 2 merge BH 1
+         else                                    merge_index = 2;
+         CM_BH_Mass[0] += CM_BH_Mass[1];
+         CM_BH_Mass[1] = 0.0;
+
+//       relabel the BH and DM particles being merged
+         CM_ClusterIdx_Cur[1] = 0;
+         for (long p=0; p<amr->Par->NPar_AcPlusInac; p++)
+         {
+            if ( amr->Par->AttributeInt[Idx_ParHalo][p] != (long_par)1 )   continue;
+            if ( amr->Par->Mass[p] < (real)0.0 )   continue;
+
+            amr->Par->Type[p] = PTYPE_DARK_MATTER;
+         } // for (long p=0; p<amr->Par->NPar_AcPlusInac; p++)
+         Aux_Message( stdout, "BHs Merge! In rank %d, TimeNew = %14.8e; merge_index = %d, "
+                              "BHPos1 = %14.8e, %14.8e, %14.8e; BHPos2 = %14.8e, %14.8e, %14.8e; "
+                              "BHVel1 = %14.8e, %14.8e, %14.8e; BHVel2 = %14.8e, %14.8e, %14.8e; "
+                              "AbsRelPos = %14.8e, AbsRelVel = %14.8e, escape_vel = %14.8e.\n",
+                      MPI_Rank, TimeNew, merge_index,
+                      CM_BH_Pos[0][0], CM_BH_Pos[0][1], CM_BH_Pos[0][2], CM_BH_Pos[1][0], CM_BH_Pos[1][1], CM_BH_Pos[1][2],
+                      CM_BH_Vel[0][0], CM_BH_Vel[0][1], CM_BH_Vel[0][2], CM_BH_Vel[1][0], CM_BH_Vel[1][1], CM_BH_Vel[1][2],
+                      AbsRelPos, AbsRelVel, escape_vel );
+      } // if ( AbsRelPos < R_acc  &&  AbsRelVel < 3*escape_vel )
+   } // if ( Merger_Coll_NumBHs == 2 )
+
+
+   if ( AGN_feedback )
+   {
+//    (3) set the injection parameters
+//    set the jet direction vector
+      SetJetDirection( TimeNew );
```

Comment:
`SetJetDirection()` is called before computing `ang_mom_sum`. This will lead to an incorrect jet direction in the first step and change results after restart.